### PR TITLE
DS-3856 - foreignkey-constraint community2community_child_comm_id_fkey

### PR DESCRIPTION
**Pull request by fritzsche-stefan**
https://jira.duraspace.org/browse/DS-3856

I am not sure what will happen if an error occurs in the rawDelete method. The child's parent relationship may be lost.

### CODE REVIEW

**Comment by tdonohue** [comment]
@antzsch : thanks for the PR.  According to Travis CI, this small change results in Integration Test failures.  The error is:
```
testCommunityAdminDeletions(org.dspace.content.ITCommunityCollection)  Time elapsed: 0.366 sec  <<< ERROR!
org.databene.contiperf.PerfTestExecutionError: org.dspace.authorize.AuthorizeException: Authorization denied for action WRITE on COMMUNITY:26bc9034-a708-481b-817b-c7c146380113 by user 8adc399e-e3e2-4085-9706-6a60138ef9c3
	at org.dspace.content.ITCommunityCollection.testCommunityAdminDeletions(ITCommunityCollection.java:248)
```

That error seems to be saying that our ITCommunityCollection test (line 248) is now throwing errors, because the Community Administrator no longer has rights to delete SubCommunities of Communities that they are in charge of.  Here's that line 248 in the test class: https://github.com/DSpace/DSpace/blob/dspace-6_x/dspace-api/src/test/java/org/dspace/content/ITCommunityCollection.java#L248

So, it looks like this solution may fix the bug you are seeing, but causes further issues with Community Administrator permissions.  The reason this occurs now is likely because the child community is now detached from its parent first, and then `rawDelete()` is called.  That means the `rawDelete()` no longer knows where this child community exists in the hierarchy, and who has permissions to delete it.

My best guess here is that this needs further analysis / work. It doesn't look like we can merge this code as-is, since it breaks permission inheritance for Community Administrators.

**Review by tdonohue — APPROVED** [review]
👍 Code looks reasonable. The actual fix looks like a minor code refactor to `removeSubcommunity()`, and that refactor seems reasonable.

I would like verification that the newly added `ITCommunityCollection` test *fails* before this refactor is applied.  It sounds like it should, but might be worth a quick test here to ensure the new test is properly testing the changed code.

**Comment by jsavell** [comment]
This fix worked fine for me.

**Comment by tdonohue** [comment]
Thanks for the review/test @jsavell! 

@kshepherd : this looks like it's likely good to merge immediately, pending your final approval (looks like we just need one more committer to vote +1).

**Review by kshepherd — APPROVED** [review]

**Comment by kshepherd** [comment]
Merging now, thanks for the contribution @antzsch !
This will likely need porting to master branch so I've added the "port to master" label here to remind ourselves

**Pull request merged**
